lib/Dialect/ArithExt/IR/ArithExtOps.td

```diff
@@ -16,8 +16,9 @@ class ArithExt_Op<string mnemonic, list<Trait> traits = [Pure]> :
 
 class ArithExt_BinaryOp<string mnemonic, list<Trait> traits = []> :
     ArithExt_Op<mnemonic, traits # [SameOperandsAndResultType, Pure, ElementwiseMappable]>,
-    Arguments<(ins SignlessIntegerLike:$lhs, SignlessIntegerLike:$rhs, I64Attr:$modulus)>,
+    Arguments<(ins SignlessIntegerLike:$lhs, SignlessIntegerLike:$rhs, APIntAttr:$modulus)>,
```

Comment:
I never quite understood why APIntAttr exists, when it seems to be equivalent to `Builtin_IntegerAttr`, which requires an explicit specification of the integer container bit width. I suspect the difference is that APIntAttr is not a typed attribute, and I'm not sure which is better for our purposes.

Reply:
I guess it doesn't matter as long as the operands are typed, and the modulus fits within those operands.

Reply:
I think `Builtin_IntegerAttr ` isthe actual tablegen definition (autogeneration source) of `mlir::IntegerAttr`, whereas `APIntAttr` is actually a `Constraint` (basically `CPred<isa<mlir::IntegerAttr>($_self)>`, iirc.). 

I'm not sure `Builtin_IntegerAttr` would be valid in a place where a constraint is expected? Similarily, `APIntAttr` is, afaik, a purely TableGen concept and has no C++ equivalent (I guess, `isa<mlir::IntegerAttr>` is the equivalant thing). Anytime I interact with MLIR's integer type/attribute system, I get a mild headache ;)

Reply:
We use Builtin_IntegerAttr elsewhere in HEIR, and it requires a type specification that corresponds to the chosen bitwidth of the APInt that stores it.

Reply:
Interesting! I guess it does make sense that one can use `TypeOrAttrDef`s directly as a constraint, though I haven't seen `Builtin_IntegerAttr` used that way upstream yet. 

I tried swapping ut `APIntAttr` for `Builtin_IntegerAttr`  to see what happens, but before I even got to see what that does to the textual representation, I noticed that `getModulus()` now returns the Attr directly, whereas with `APIntAttr`, the tablegen backend would have that function name be a helper function that returns an `APInt`, and demote the actual `Attr` getter to `getModulusAttr()`. I'd love to say I understand way, but I'm at "MLIR works in mysterious ways" 🤷‍♂️